Framework/Reflectometry/src/ReflectometryReductionOneAuto3.cpp

```diff
@@ -842,144 +853,163 @@ bool ReflectometryReductionOneAuto3::processGroups() {
+  // workspace names and output properties
+  std::vector<std::string> IvsQBinnedGroup, IvsQGroup, IvsLamGroup;
+  OutputProperties outputProperties;
+  for (size_t i = 0; i < group->size(); ++i) {
+    auto inputName = group->getItem(i)->getName();
+    auto outputNames = getOutputNamesForGroupMember(inputName, runNumber, i);
+    auto alg = createAlgorithmForWorkspaceInGroup(inputName, outputNames);
+    alg->execute();
+
+    // Cache the output names
+    IvsQGroup.emplace_back(outputNames.iVsQ);
+    IvsQBinnedGroup.emplace_back(outputNames.iVsQBinned);
+    IvsLamGroup.emplace_back(outputNames.iVsLam);
+
+    // Cache the other output properties so that we can set them as our outputs
+    // (required by the GUI). We can only display one set on the GUI so just
+    // return the last set processed
+    outputProperties.thetaIn = alg->getProperty("ThetaIn");
+    outputProperties.qMin = alg->getProperty("MomentumTransferMin");
+    outputProperties.qMax = alg->getProperty("MomentumTransferMax");
+    outputProperties.qStep = alg->getProperty("MomentumTransferStep");
+    outputProperties.scale = alg->getProperty("ScaleFactor");
+  }
+
+  // Set the main algorithm output properties (workspaces and other calculated
+  // values)
+  const auto outputGroupNames = getOutputWorkspaceNames();
+  setOutputProperties(outputProperties);
+  setOutputWorkspaces(IvsQGroup, IvsQBinnedGroup, IvsLamGroup,
+                      outputGroupNames);
+
+  const bool polarizationAnalysisOn = getProperty("PolarizationAnalysis");
   if (!polarizationAnalysisOn) {
     // No polarization analysis. Reduction stops here
     return true;
   }
 
-  applyPolarizationCorrection(output.iVsLam);
+  applyPolarizationCorrection(outputGroupNames.iVsLam);
 
   // Polarization correction may have changed the number of workspaces in the
   // groups
   IvsLamGroup.clear();
   IvsQBinnedGroup.clear();
   IvsQGroup.clear();
 
-  // Now we've overwritten the IvsLam workspaces, we'll need to recalculate
-  // the IvsQ ones
-  alg->setProperty("FirstTransmissionRun", "");
-  alg->setProperty("SecondTransmissionRun", "");
-  alg->setProperty("CorrectionAlgorithm", "None");
-
-  auto outputIvsLamNames = workspaceNamesInGroup(output.iVsLam);
-  for (size_t i = 0; i < outputIvsLamNames.size(); ++i) {
-    auto inputName = group->getItem(i)->getName();
-    auto outputNames = getOutputNamesForGroups(inputName, runNumber, i);
-
-    const std::string IvsQName = outputNames.iVsQ;
-    const std::string IvsQBinnedName = outputNames.iVsQBinned;
-    const std::string IvsLamName = outputIvsLamNames[i];
-
-    // Find the spectrum processing instructions for ws index 0
+  // Now we've overwritten the IvsLam workspaces, we'll need to recalculate the
+  // IvsQ ones
+  auto correctedIvsLamWorkspaces =
+      workspaceNamesInGroup(outputGroupNames.iVsLam);
+  for (size_t i = 0; i < correctedIvsLamWorkspaces.size(); ++i) {
```

Comment:
... `void recalculateGroupIvsq(...)` since it appears that it's mostly member access in both loops anyway